src/functional/swing_slide_filter.zig

```diff
@@ -0,0 +1,411 @@
+                    uncompressed_values[current_segment.start_time],
+                );
+                const end_value: f64 = evaluateLineAtTime(current_line, current_timestamp - 1);
+                try appendValue(end_value, compressed_values);
+            } else {
+                try appendValue(current_segment.end_value, compressed_values);
+            }
+
+            try appendIndex(current_timestamp, compressed_values);
+
+            // Update the current segment.
+            current_segment.start_time = current_timestamp;
+            current_segment.start_value = uncompressed_values[current_timestamp];
+
+            // Catch edge case (only one point left).
+            // If `current_timestamp+1 >= uncompressed_values.len`
+            // then `uncompressed_values[current_timestamp + 1]` will return index out of bound error.
+            if (current_timestamp + 1 < uncompressed_values.len) {
+                current_segment.end_time = current_timestamp + 1;
+                current_segment.end_value = uncompressed_values[current_timestamp + 1];
+
+                updateSwingLine(current_segment, &upper_bound_line, adjusted_error_bound);
+                updateSwingLine(current_segment, &lower_bound_line, -adjusted_error_bound);
+
+                current_timestamp += 1;
+                slope_derivate = computeSlopeDerivate(current_segment);
+            } else { // Create line with slope 0 and intercept equal to the current value.
+                current_segment.end_time = current_timestamp;
+                current_segment.end_value = uncompressed_values[current_timestamp];
+                upper_bound_line.slope = 0.0;
+                upper_bound_line.intercept = uncompressed_values[current_timestamp];
+                lower_bound_line.slope = 0.0;
+                lower_bound_line.intercept = uncompressed_values[current_timestamp];
+            }
+        } else { //Filtering mechanism (the current point is still inside the limits).
+
+            current_segment.end_time = current_timestamp;
+            current_segment.end_value = uncompressed_values[current_timestamp];
+
+            // Update the potentially new upper and lower bounds with the new current point.
+            updateSwingLine(current_segment, &new_upper_bound_line, adjusted_error_bound);
+            updateSwingLine(current_segment, &new_lower_bound_line, -adjusted_error_bound);
+
+            const new_upper_limit: f80 = evaluateLineAtTime(new_upper_bound_line, current_timestamp);
+            const new_lower_limit: f80 = evaluateLineAtTime(new_lower_bound_line, current_timestamp);
+
+            // Update the upper and lower bounds if needed.
+            if (upper_limit > new_upper_limit) { // Swing down.
+                upper_bound_line.slope = new_upper_bound_line.slope;
+                upper_bound_line.intercept = new_upper_bound_line.intercept;
+            }
+            if (lower_limit < new_lower_limit) { //Swing up.
```

Comment:
Maybe make Line 207 only contain the comment for consistency with regards to placement of comments.

Reply:
fixed.